### DISCUSSION TITLE
Exclude route from dependencies

### DISCUSSION THREAD

**Discussion by nodarai** [fastapi/fastapi#3395] [Questions] [answered]
Hello,

If I'm creating a router with the following code 
```python
APIRouter(prefix="/apps", dependencies=[Depends(require_token)])
```
which requires a token for all the routes in that router.
How can I exclude a route from the dependencies, so that a token is not required to use that route?

I know that I can set dependency directly on the route, bit if I have 10+ routes and I only want to exclude one it's not very useful to manually set the dependencies on 10+ routes.
I have also tried setting route's dependency to `[]` but it does not work.

**Comment by r-bar** [comment, marked as answer]
I would probably use 2 separate routers, then include them both in your app.

```py
app = FastAPI()
router = APIRouter(prefix="/apps", dependencies=[Depends(require_token)])
unauthenticated_router = APIRouter(prefix="/apps")

@router.get('/')
def normal_route():
    return

@unauthenticated_router.get('/public')
def unauthenticated_route():
    return

app.include_router(router)
app.include_router(unauthenticated_router)
```

**Reply by nodarai** [reply]
Thanks for the solution.
But it feels more like a workaround rather then a feature :slightly_smiling_face:
